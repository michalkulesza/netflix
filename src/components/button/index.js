import React from "react";
import { Main, Square, Round, Container, Label, Clear, Form } from "./styles/button";

import { ImSpinner8 } from "react-icons/im";

const Button = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Button.Square = ({ iconScale, inverted, marginRight, children, ...restProps }) => {
	return (
		<Square iconScale={iconScale} inverted={inverted} marginRight={marginRight} {...restProps}>
			{children}
		</Square>
	);
};

Button.Round = ({ label, inverted, dark, marginRight, children, isLoading, ...restProps }) => {
	return (
		<Container marginRight={marginRight}>
			<Round inverted={inverted} dark={dark} isLoading={isLoading} {...restProps}>
				{children}
			</Round>
			{label && <Label>{label}</Label>}
		</Container>
	);
};

Button.Clear = ({ size, padding, children, margin, ...restProps }) => {
	return (
		<Clear size={size} padding={padding} margin={margin} {...restProps}>
			{children}
		</Clear>
	);
};

Button.Form = ({ loading, children, ...restProps }) => {
	return <Form {...restProps}>{loading ? <ImSpinner8 /> : children}</Form>;
};

export default Button;
