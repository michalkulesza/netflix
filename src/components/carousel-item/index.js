import React from "react";
import { Item, Container, Wrapper, Loader, ExpandedSmall } from "./styles/carousel-item";

const CarouselItem = ({ children, ...restProps }) => {
	return <Item {...restProps}>{children}</Item>;
};

CarouselItem.Container = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

CarouselItem.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

CarouselItem.Loader = ({ children, ...restProps }) => {
	return <Loader {...restProps}>{children}</Loader>;
};

CarouselItem.ExpandedSmall = ({ children, ...restProps }) => {
	return <ExpandedSmall {...restProps}>{children}</ExpandedSmall>;
};

export default CarouselItem;
