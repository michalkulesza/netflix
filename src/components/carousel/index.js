import React from "react";
import {
	Main,
	Category,
	Overlay,
	Button,
	Container,
	Wrapper,
	ItemsContainer,
	ItemWrapper,
	Item,
	Loader,
} from "./styles/carousel";

const Carousel = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Carousel.Category = ({ children, ...restProps }) => {
	return <Category {...restProps}>{children}</Category>;
};

Carousel.Overlay = ({ children, ...restProps }) => {
	return <Overlay {...restProps}>{children}</Overlay>;
};

Carousel.Button = ({ children, ...restProps }) => {
	return <Button {...restProps}>{children}</Button>;
};

Carousel.Container = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

Carousel.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

Carousel.ItemsContainer = ({ children, ...restProps }) => {
	return <ItemsContainer {...restProps}>{children}</ItemsContainer>;
};

Carousel.ItemWrapper = ({ children, ...restProps }) => {
	return <ItemWrapper {...restProps}>{children}</ItemWrapper>;
};

Carousel.Item = ({ children, ...restProps }) => {
	return <Item {...restProps}>{children}</Item>;
};

Carousel.Loader = ({ children, ...restProps }) => {
	return <Loader {...restProps}>{children}</Loader>;
};

export default Carousel;
