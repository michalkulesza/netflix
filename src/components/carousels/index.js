import React from "react";
import { Main } from "./styles/carousels";

const Carousels = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

export default Carousels;
