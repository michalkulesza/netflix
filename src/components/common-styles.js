import { keyframes } from "styled-components/macro";

export const LoadingAnimation = keyframes`
	from {
		background-position-x: 0vw;
	}
	to {
		background-position-x: 100vw;
	}
`;
