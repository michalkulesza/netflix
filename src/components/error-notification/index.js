import React from "react";
import { Main, Wrapper } from "./styles/error-notification";

const ErrorNotification = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

ErrorNotification.Wrapper = ({ children, ...restProps }) => {
	return <Wrapper {...restProps}>{children}</Wrapper>;
};

export default ErrorNotification;
