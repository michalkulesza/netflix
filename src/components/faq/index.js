import React from "react";
import { Main, Container, Title, List, Item, Header, Icon, Body, Text } from "./styles/faq";

const Faq = ({ children, ...restProps }) => {
	return <Main {...restProps}>{children}</Main>;
};

Faq.Container = function ({ children, ...restProps }) {
	return <Container {...restProps}>{children}</Container>;
};

Faq.Title = function ({ children, ...restProps }) {
	return <Title {...restProps}>{children}</Title>;
};

Faq.List = function ({ children, ...restProps }) {
	return <List {...restProps}>{children}</List>;
};

Faq.Item = function ({ children, ...restProps }) {
	return <Item {...restProps}>{children}</Item>;
};

Faq.Header = function ({ children, ...restProps }) {
	return <Header {...restProps}>{children}</Header>;
};

Faq.Icon = function ({ children, ...restProps }) {
	return <Icon {...restProps}>{children}</Icon>;
};

Faq.Body = function ({ children, ...restProps }) {
	return <Body {...restProps}>{children}</Body>;
};

Faq.Text = function ({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

export default Faq;
