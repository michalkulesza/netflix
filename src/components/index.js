export { default as Header } from "./header";
export { default as Navbar } from "./navbar";
export { default as Jumbotron } from "./jumbotron";
export { default as Faq } from "./faq";
export { default as Newsletter } from "./newsletter";
export { default as Footer } from "./footer";
export { default as Button } from "./button";
export { default as Form } from "./form";
export { default as Carousel } from "./carousel";
export { default as Carousels } from "./carousels";
export { default as Item } from "./item";
export { default as ItemExpanded } from "./item-expanded";
export { default as Details } from "./item-details/details";

export { default as ItemDetailsHeader } from "./item-details-header";
export { default as ItemDetailsInfo } from "./item-details-info";
export { ItemDetailsEpisodes } from "./item-details-episodes";
export { ItemDetailsEpisode } from "./item-details-episodes";
