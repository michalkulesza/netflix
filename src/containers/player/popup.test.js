import React from "react";
import { shallow } from "enzyme";
import PopupContainer from "./popup";
import { Popup } from "../../components";

describe("<PopupContainer/> with no props", () => {
	const wrapper = shallow(<PopupContainer></PopupContainer>);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should not render children", () => {
		expect(wrapper.find(Popup.Dialog).children()).toHaveLength(0);
	});
});

describe("<PopupContainer/> with no props", () => {
	const wrapper = shallow(
		<PopupContainer>
			<span>testElem</span>
			<span>testElem</span>
			<span>testElem</span>
		</PopupContainer>
	);

	it("should match the snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should not render children", () => {
		expect(wrapper.find(Popup.Dialog).children()).toHaveLength(3);
	});
});
