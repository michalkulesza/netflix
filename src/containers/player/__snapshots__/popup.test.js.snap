// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupContainer/> with no props should match the snapshot 1`] = `"<div class=\\"popup__Main-al6sl5-0 hIqMKL\\"><div class=\\"popup__Dialog-al6sl5-1 iSKUtS\\"></div></div>"`;

exports[`<PopupContainer/> with no props should match the snapshot 2`] = `"<div class=\\"popup__Main-al6sl5-0 hIqMKL\\"><div class=\\"popup__Dialog-al6sl5-1 iSKUtS\\"><span>testElem</span><span>testElem</span><span>testElem</span></div></div>"`;
